addons/hr_timesheet/hr_timesheet_view.xml

```diff
@@ -51,23 +104,39 @@
                     <field name="project_id"/>
                     <field name="task_id"/>
                     <field name="name"/>
-                    <field name="department_id"/>
+                    <filter name="mine" string="My Timesheets" domain="[('user_id', '=', uid)]"/>
+                    <separator/>
+                    <filter name="week" string="This Week"
+                            domain="[
+                                '&amp;',
+                                ('date', '>=', (context_today() + relativedelta(weeks=-1,days=1,weekday=0)).strftime('%Y-%m-%d')),
+                                ('date', '&lt;=', (context_today() + relativedelta(weekday=6)).strftime('%Y-%m-%d')),
```

Comment:
> Shouldn't "this week" really be "this week" instead of between today and next day of next week ?

Should be what this does, "weekday" is the DOW (next one by default) with 0 = monday and 6 = sunday. So the domain is between "previous monday" and "next sunday".